frontend/src/employee-frontend/components/child-information/ChildDiscussions.tsx

```diff
@@ -0,0 +1,224 @@
+// SPDX-FileCopyrightText: 2017-2023 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import { faPen } from 'Icons'
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+import LocalDate from 'lib-common/local-date'
+import { useMutationResult, useQueryResult } from 'lib-common/query'
+import { UUID } from 'lib-common/types'
+import Title from 'lib-components/atoms/Title'
+import AsyncButton from 'lib-components/atoms/buttons/AsyncButton'
+import Button from 'lib-components/atoms/buttons/Button'
+import InlineButton from 'lib-components/atoms/buttons/InlineButton'
+import { FixedSpaceRow } from 'lib-components/layout/flex-helpers'
+import DatePicker from 'lib-components/molecules/date-picker/DatePicker'
+import { fontWeights } from 'lib-components/typography'
+
+import { useTranslation } from '../../state/i18n'
+import { UIContext, UiState } from '../../state/ui'
+import { RequireRole } from '../../utils/roles'
+import { FlexRow } from '../common/styled/containers'
+
+import {
+  childDiscussionQuery,
+  createChildDiscussionMutation,
+  updateChildDiscussionMutation
+} from './queries'
+
+interface Props {
+  childId: UUID
+}
+
+const DataRow = styled.div`
+  display: flex;
+  min-height: 2rem;
+`
+
+const DataLabel = styled.div`
+  width: 240px;
+  padding: 0 40px 0 0;
+  margin: 0;
+  font-weight: ${fontWeights.semibold};
+`
+
+const DataValue = styled.div`
+  display: flex;
+`
+
+const RightAlignedRow = styled.div`
+  display: flex;
+  align-items: baseline;
+  justify-content: flex-end;
+  padding-top: 20px;
+`
+
+export default React.memo(function ChildDiscussions({ childId }: Props) {
+  const { i18n, lang } = useTranslation()
+
+  const [offeredDate, setOfferedDate] = useState<LocalDate | null>(null)
+  const [heldDate, setHeldDate] = useState<LocalDate | null>(null)
+  const [counselingDate, setCounselingDate] = useState<LocalDate | null>(null)
+
+  const childDiscussionData = useQueryResult(childDiscussionQuery(childId))
+
+  useEffect(() => {
+    if (!childDiscussionData.isSuccess) return
+
+    setOfferedDate(childDiscussionData.value.offeredDate)
+    setHeldDate(childDiscussionData.value.heldDate)
+    setCounselingDate(childDiscussionData.value.counselingDate)
+  }, [childDiscussionData])
```

Comment:
Don't use `useEffect` for initializing component state. Instead, create an outer component that does the query and uses `renderResult()` to render an inner component that is passed the initial data as props and renders the view/edit stuff. It will probably also simplify the view/edit mode handling.

See https://react.dev/learn/you-might-not-need-an-effect for more info about avoiding `useEffect`.

Reply:
You might also benefit from using eVaka's "form framework" which has some learning curve but greatly reduces the need for manual state management and validation. See [here](https://github.com/espoon-voltti/evaka/blob/master/frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx#L312-L328) for an example.

Reply:
Interesting to use an outer component, sounds like a good approach.

I'll add the outer component for initialization and check out the form framework.